### PR TITLE
simplify(coverage)

### PR DESCRIPTION
**Pull request by FlyCloudC**
- simplifies the implementation
- add a test for empty case

### CODE REVIEW

File: coverage/coverage.mbt [hunk 1]
```diff
@@ -16,73 +16,43 @@
...
-  logger.write_char('[')
-  for i in 0..<self.counter.length() {
-    if i != 0 {
-      logger.write_string(", ")
-    }
-    Show::output(self.counter[i], logger)
-  }
-  logger.write_char(']')
+  self.0.output(logger)
 }
```
Comment (lynzrand) [inline coverage/coverage.mbt]: This puts a dependency on the `Show` implementation of `(Array[Int] as Show)::output` being the same as JSON array's format. Although we are enforcing it via the tests below, shouldn't we at least add a note or something :)

**Review by lynzrand — APPROVED** [review]
LGTM as long as the output format doesn't change.

File: coverage/coverage.mbt (hunk 1 above)
Comment (FlyCloudC) [reply coverage/coverage.mbt]: done

**Comment by bobzhang** [comment]
this package was supposed to have minimal deps, let's not change it unless we have clear benefits

**Comment by lynzrand** [comment]
> this package was supposed to have minimal deps, let's not change it unless we have clear benefits

AFAICT this does not change the dependency list of the package, but still makes it somewhat more readable?

---

Okay, maybe this comment is referring to the `Array::show` thing, which, in that case, I consider both sides reasonable -- rely less on "current" stdlib behavior that we don't have a guarantee on is definitely a good thing. The other parts are still reasonable changes, though.

**Comment by peter-jerry-ye** [comment]
We will only accept the first commit ebdf4b506d508ad1e473b5774c925467754cbcce.

**Comment by FlyCloudC** [comment]
> We will only accept the first commit ebdf4b506d508ad1e473b5774c925467754cbcce.

Should I submit a revert commit on this branch?

**Comment by peter-jerry-ye** [comment]
> Should I submit a revert commit on this branch?

I think a simple `git rebase -i main` should do the trick to drop the other commits.
